### DISCUSSION TITLE
Avoid max and min check when input is disabled

### DISCUSSION THREAD

**Discussion by giacomomasseron** [filamentphp/filament#14469] [Help]
### Package

Panel builder

### Package Version

v3.2.115

### How can we help you?

Hi,  
I need to avoid max and min check when input is disabled.

I have this `TextInput`:

	Components\TextInput::make('quantity')
		->label(__('Quantity'))
		->maxValue(function (Get $get) {
			return $get('max_quantity');
		})
		->disabledOn('edit')
		->afterStateUpdated(function (Get $get, Set $set, $state) {
			if ($state > $get('max_quantity')) {
				$set('quantity', $get('max_quantity'));
			}
		})
		->numeric()
		->required(),

Even if the input is disabled, I get the error message about "max_quantity".

I would use `when`, but I did not understand how:

	Components\TextInput::make('quantity')
		->label(__('Quantity'))
        ->when(WE_ARE_IN_EDIT, function ($component) {
            $component->maxValue(function (Get $get) {
			    return $get('max_quantity');
		    });
        })
		->disabledOn('edit')
		->afterStateUpdated(function (Get $get, Set $set, $state) {
			if ($state > $get('max_quantity')) {
				$set('quantity', $get('max_quantity'));
			}
		})
		->numeric()
		->required(),

**Comment by ronssij** [comment]
@glennjacobs I think this should work.

```php
->maxValue(function (Component $livewire, Get $get) {
     return $livewire->isDisabled() ? 0 : $get('max_quantity');
})
```
